packages/gafl-webapp-service/src/processors/__tests__/licence-type-display.spec.js

```diff
@@ -0,0 +1,99 @@
+import { hasJunior, hasSenior } from '../concession-helper.js'
+import { licenceTypeDisplay, licenceTypeAndLengthDisplay, isPhysical } from '../licence-type-display.js'
+
+const permission = {}
+
+const mssgs = {
+  over_65: 'Over 65',
+  age_junior: 'Junior',
+  licence_type_radio_salmon: 'Salmon and sea trout',
+  licence_type_radio_trout_two_rod: 'Trout and coarse, up to 2 rods',
+  licence_type_radio_trout_three_rod: 'Trout and coarse, up to 3 rods'
+}
+
+jest.mock('../concession-helper', () => ({
+  hasJunior: jest.fn(),
+  hasSenior: jest.fn()
+}))
+
+jest.mock('../mapping-constants', () => ({
+  LICENCE_TYPE: {
+    'trout-and-coarse': 'Trout and coarse',
+    'salmon-and-sea-trout': 'Salmon and sea trout'
+  }
+}))
+
+describe('licenceTypeDisplay', () => {
+  it('returns junior if person is junior', () => {
+    hasJunior.mockImplementationOnce(() => true)
+    permission.licenceType = 'Salmon and sea trout'
+    const result = licenceTypeDisplay(permission, mssgs)
+    expect(result).toEqual('Junior, Salmon and sea trout')
+  })
+
```

Comment:
As discussed, it might be worth creating an array of possible permissions in each test. For example:
```
it ([
['Junior', 'salmon and sea trout', '2 rod'],
['Junior', 'salmon and sea trout', '3 rod'] etc_
```
So all different scenarios are tested for each concession.